apps/vaults/components/ListHero.tsx

```diff
@@ -53,7 +53,7 @@ export function ListHero({
 							const selectedChains = options
 								.filter((o): boolean => o.isSelected)
 								.map((option): number => Number(option.value));
-							onChangeChains(selectedChains);
+							onChangeChains(selectedChains.length === 0 ? undefined : selectedChains);
```

Comment:
I'm not sure this is a good change, adds complexity for no apparent reason.

I would try and have the empty array, or why is that not possible and we need to go with `undefined`?